### PR TITLE
Add noise_seed/photon_target

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: translators/scene.py [hunk 1]
```diff
@@ -647,13 +647,17 @@ def __translate_frame(self):
             width = int(self.bl_scene.render.resolution_x * scale)
             height = int(self.bl_scene.render.resolution_y * scale)
 
+        noise_seed = asr_scene_props.noise_seed + self.bl_scene.frame_current \
```
Comment (dictoon) [inline translators/scene.py]: I suggest adding parentheses to make operator precedence unambiguous. It's not obvious to me right now how this expression is interpreted.

Comment (dictoon) [reply translators/scene.py]: It's interesting that you're still considering the noise seed even when "Vary by Frame" is checked.

File: properties/scene.py [hunk 2]
```diff
@@ -199,6 +199,14 @@ class AppleseedRenderSettings(bpy.types.PropertyGroup):
                                           description="Automatically determine the number of rendering threads",
                                           default=True)
 
+    noise_seed = bpy.props.IntProperty(name="noise_seed",
+                                       default=0)
+
+    per_frame_noise = bpy.props.BoolProperty(name="per_frame_noise",
+                                             description="Vary the noise seed by the frame number.\n"
+                                                         "This gives better results during animations.",
```
Comment (dictoon) [inline properties/scene.py]: *in* animations maybe?

File: translators/camera.py [hunk 3]
```diff
@@ -137,7 +137,7 @@ def __set_params(self, scene):
     def __ortho_camera_params(self, scene, aspect_ratio):
         camera = self.bl_camera.data
         cam_params = {'aspect_ratio': aspect_ratio,
-                      'near_z': camera.appleseed.near_z,
+                      'near_z': camera.clip_start,
```
Comment (dictoon) [inline translators/camera.py]: Are you sure that `camera.clip_start` is negative? In appleseed, Z+ goes out of the screen (toward the user) while Z- goes into the screen (away from the user). In the local space of the camera, we want the near Z plane to be slightly away from the origin along Z-, hence it must be a negative value.

File: ui/__init__.py [hunk 4]
```diff
@@ -157,6 +115,9 @@ def register():
     #    particles.register()
     lamps.register()
 
+    for panel in get_panels():
+        panel.COMPAT_ENGINES.add('APPLESEED_RENDER')
```
Comment (dictoon) [inline ui/__init__.py]: Doesn't this need to be undone in `unregister()`?

File: ui/textures.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
+
+#
+# This source file is part of appleseed.
+# Visit https://appleseedhq.net/ for additional information and resources.
+#
+# This software is released under the MIT license.
+#
+# Copyright (c) 2014-2018 Jonathan Dent, The appleseedhq Organization
```
Comment (dictoon) [inline ui/textures.py]: New file hence notice should only contain today's date.

Maybe at some point we'll revisit and simplify our rules for copyright notices...

File: ui/textures.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            if tex:
+                split = layout.split(percentage=0.2)
+                split.label(text="Type:")
+                split.prop(tex, "type", text="")
+
+
+class AppleseedTextureMapping(AppleseedTexturePanelBase):
+    bl_label = "Mapping"
+    bl_context = "texture"
+
```
Comment (dictoon) [inline ui/textures.py]: Remove one blank line?

File: ui/textures.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+    def draw(self, context):
+        layout = self.layout
+
```
Comment (dictoon) [inline ui/textures.py]: Remove blank line?

File: ui/textures.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        node = context.texture_node
+
```
Comment (dictoon) [inline ui/textures.py]: Remove blank line?

File: ui/textures.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+def register():
+    util.safe_register_class(AppleseedTexturePanelBase)
+    util.safe_register_class(AppleseedTextureContext)
+    util.safe_register_class(AppleseedTextureMapping)
+
+
+def unregister():
+    util.safe_unregister_class(AppleseedTextureMapping)
+    util.safe_unregister_class(AppleseedTextureContext)
+    util.safe_unregister_class(AppleseedTexturePanelBase)
```
Comment (dictoon) [inline ui/textures.py]: Make sure the last line ends with an end-of-line character.

File: translators/scene.py (hunk 1 above)
Comment (jdent02) [reply translators/scene.py]: Is that not the way to do it?

Comment (dictoon) [reply translators/scene.py]: I don't know if it's _the_ way, in any case it's better than what I had in mind.

**Comment by jdent02** [comment]
Everything's fixed I think

File: translators/camera.py [hunk 10]
```diff
@@ -137,7 +137,7 @@ def __set_params(self, scene):
     def __ortho_camera_params(self, scene, aspect_ratio):
         camera = self.bl_camera.data
         cam_params = {'aspect_ratio': aspect_ratio,
-                      'near_z': camera.clip_start,
+                      'near_z': camera.appleseed.near_z,
```
Comment (dictoon) [inline translators/camera.py]: Ok, that's one way to do it, however I wasn't against using `camera.clip_start` (but negated) if it makes sense...

File: translators/scene.py [hunk 11]
```diff
@@ -625,7 +625,10 @@ def __translate_render_settings(self):
 
         if asr_scene_props.shading_override:
             parameters['shading_engine'] = {'override_shading': {'mode': asr_scene_props.override_mode}}
+            
         conf_final.set_parameters(parameters)
+
+        parameters['lighting_engine'] = 'pt'
```
Comment (dictoon) [inline translators/scene.py]: That looks a bit like a hack :)

I don't quite understand how this meshes with

    lighting_engine = asr_scene_props.lighting_engine if self.export_mode != ProjectExportMode.INTERACTIVE_RENDER else 'pt'

a few lines above.

Moreover, all settings (PT *and* SPPM) should always be exported, even when just one engine is selected. That allows for instance to change the render engine in appleseed.studio and still benefit from the parameters set in Blender. Let me know if I'm not clear, it's early morning here :)

On an unrelated note, the following code (lines 604-608) can be set via the dictionary like other parameters:

```
parameters['pt']['max_diffuse_bounces'] = asr_scene_props.max_diffuse_bounces if not asr_scene_props.max_diffuse_bounces_unlimited else -1
parameters['pt']['max_glossy_bounces'] = asr_scene_props.max_glossy_brdf_bounces if not asr_scene_props.max_glossy_brdf_bounces_unlimited else -1	             parameters['pt']['max_glossy_bounces'] = asr_scene_props.max_glossy_brdf_bounces if not asr_scene_props.max_glossy_brdf_bounces_unlimited else -1
parameters['pt']['max_specular_bounces'] = asr_scene_props.max_specular_bounces if not asr_scene_props.max_specular_bounces_unlimited else -1	             parameters['pt']['max_specular_bounces'] = asr_scene_props.max_specular_bounces if not asr_scene_props.max_specular_bounces_unlimited else -1
parameters['pt']['max_volume_bounces'] = asr_scene_props.max_volume_bounces if not asr_scene_props.max_volume_bounces_unlimited else -1	             parameters['pt']['max_volume_bounces'] = asr_scene_props.max_volume_bounces if not asr_scene_props.max_volume_bounces_unlimited else -1
parameters['pt']['max_bounces'] = asr_scene_props.max_bounces if not asr_scene_props.max_bounces_unlimited else -1	             parameters['pt']['max_bounces'] = asr_scene_props.max_bounces if not asr_scene_props.max_bounces_unlimited else -1
```

Comment (dictoon) [reply translators/scene.py]: Ok, thanks.

**Comment by dictoon** [comment]
What's the status of this? Do you have more changes planned @jdent02 or should we merge it?

**Comment by jdent02** [comment]
I do have more stuff I’m working on but this can be merged and I’ll do a new PR

**Comment by dictoon** [comment]
Ok, merging now.

Please don't forget about my remark "On an unrelated note, the following code (lines 604-608) can be set via the dictionary like other parameters" :)

**Comment by dictoon** [comment]
Ah, before I merge, can you squash the feedback commit please?

**Comment by jdent02** [comment]
I’ll do that tonight

File: translators/scene.py [hunk 12]
```diff
@@ -598,14 +598,14 @@ def __translate_render_settings(self):
                                 'volume_distance_samples': asr_scene_props.volume_distance_samples,
                                 'dl_light_samples': asr_scene_props.dl_light_samples,
                                 'ibl_env_samples': asr_scene_props.ibl_env_samples,
-                                'dl_low_light_threshold': asr_scene_props.dl_low_light_threshold}
+                                'dl_low_light_threshold': asr_scene_props.dl_low_light_threshold,
+                                'max_diffuse_bounces': asr_scene_props.max_diffuse_bounces if not asr_scene_props.max_diffuse_bounces_unlimited else -1,
+                                'max_glossy_bounces': asr_scene_props.max_glossy_brdf_bounces if not asr_scene_props.max_glossy_brdf_bounces_unlimited else -1,
```
Comment (dictoon) [inline translators/scene.py]: Note for later: rename `max_glossy_brdf_bounces`, it's the only one containing the word `brdf`.

**Pull request merged**